### DISCUSSION TITLE
Shifter improvements

### DISCUSSION THREAD

**Discussion by eMadman** [doudar/SmartSpin2k#281] [Ideas]
lack of tactile feedback on the standard shifter design was something that seemed easy to address. I think the amount of material above the button itself was the leading cause of the mushy feeling.  I tried to take some inspiration from other momentary switches and buttons I've seen and go with a thin membrane above the button.  Bridging is minimal and the most challenging part is the hole for the button to pass through - I kept this with similar geometry to the original design as a start point.  This led me here as a first workable draft.

![image](https://user-images.githubusercontent.com/24726844/146942079-8fe1a602-8422-42cb-82d5-2dbf136e6af5.png)

The ideal hit zone for the button is the flat spot at the top of the design.  I think I want this to be a bit bigger..  The 45 degree walls seem to be too stiff to offer a great spot to press, but they do flex and click the button if I use enough pressure.  With the thin and flat membrane, there isn't much interfering with the button's sound and tactility,

https://user-images.githubusercontent.com/24726844/146943831-b44025c2-b670-41b8-a407-566db250d7ce.mp4

Untested:  I think care needs to be taken for the hot glue filling - It might be easy to overfill with hot glue and end up pressing the button from behind.  While sealing, it will be important to press down on the button to make sure it has ~1mm for the click to work.

Where I'm thinking of going with this:

- Bigger hit zone
- at least 1 more top layer - 0.2 to 0.4mm thicker - top surface needs to be more water tight.
- Two buttons, 1 strap for one handed operation
- clamp mount for handlebar or a quarter turn inspired option.

**Comment by eMadman** [comment]
Reprinted with a bigger hit zone and thicker top layer (1mm now).  This seems to be the winner.  Feels nicer under the thumb and its equally easy to print.  

This barrel can easily replace what's currently on the shifter but I'm not keen on printing that and trying (limited supply of cables and switches right now).  I'll make the change and commit on a branch or my fork and then jump ahead to trying 2 switches, 1 strap for a test print later tonight.

**Reply by eMadman** [reply]
![image](https://user-images.githubusercontent.com/24726844/146997570-42740ec8-ea7f-448d-bb9d-7704e5323f78.png)

I have to re-draw the sketches and re-make the buttons but it looks like I'll have plenty of space to do something like this based on the section analysis.  

Wires would come out towards the back instead of out to the side as they do today.   I'm realizing as I type it that this might be tricky since the strap needs to bend and the tube for the wires to pass through is not a very flexible shape.

**Comment by eMadman** [comment]
This is with 5 top layers.  Trying to make this bridging watertight with TPU is turning out to be a minor pain in the butt.  I've experienced this with the topmost surfaces of the standard shifter too, but the nub is much smaller there.  This has much more flat surface bridging to deal with and the effect is more noticeable.
![image](https://user-images.githubusercontent.com/24726844/147113425-ac2c81bd-c3ee-437a-ad7a-c3e367f7e248.png)

**Comment by eMadman** [comment]
[Strap and buttons.zip](https://github.com/doudar/SmartSpin2k/files/7764252/Strap.and.buttons.zip)
two stl's here 
- one is just the buttons for rapid printing and testing - I've tested this and am somewhat happy with it.  Consider ironing on the top surface to help make it more watertight.
- one with the strap as well and ready for test printing and mounting... still a work in progress - i tried as much as possible to adapt the existing strap to avoid reinventing the wheel so there's some cleanup I need to do to the geometries there.

**Reply by doudar** [reply]
How far out are your cables and switches? Do you need me to send some? I think it’s in the documents somewhere but I use monoprice for cables.

**Reply by eMadman** [reply]
I've got a couple of extras handy that I can test with.  Some sort of mix of random Amazon and a few that I think you sent my way.  

I should be able to print and fit a unit sometime today

**Comment by eMadman** [comment]
Lesson of the day.  The shape and geometry responsible for retaining the switch to have an impact on tactile feedback.  I didn't think it would because there's just enough clearance surrounding the switch post on the "square" design to make this seem like an acceptable idea.

'The impact on tactility is quite significant.  Design on the left allows more room surrounding the post on the switch and a more directly tactile bump.  The one on the right seems to be crowding the switch and makes for a mushy, non-tactile feeling switch.

I think the one on the right offers much more stability for retaining the switch.  I'll give it one more attempt using a 6mm passthrough hole for the switch post instead of the 4mm that's pictured below. Hopefully the larger opening will result in less material insulating the switch and more clack.  I hate clack on my keyboards and use silent tactile switches for it... for the shifters, clack is king.

![20211223_103441](https://user-images.githubusercontent.com/24726844/147261969-892086ad-251e-446f-b5ca-d7355237dc93.jpg)

One last lesson:  Ironing seems to be the trick to ensuring an airtight top surface. 

Unfortunately, i've never tuned it and i certainly haven't tuned it for TPU.  Going to try ironing for solid infills only - leaving the top surface to be a concentric pattern.  Mostly because it looks nice, partly as a further contrast to the linear/monotonic solid infill in order to give one less chance for sweat to pass through

![image](https://user-images.githubusercontent.com/24726844/147263979-b3f47c3b-4f70-4f4b-b233-4a6d50f77616.png)

**Reply by doudar** [reply]
Nice journal! I love these updates.

Another thing that works well for my TPU prints is varying perimeters and infill. For instance, I’ll only do 1 or 2 top layers and maybe 2 perimeters but then print the infill at ~80-90%.

 With TPU this seems to provide a more flexible material (and doesn’t clog my nozzle as much) as printing more perimeters and top&bottom layers.

**Reply by eMadman** [reply]
I think I've got a winner here.  Printing (for me) posed a unique challenge with the watertightness.  To save print time, its definitely worth having different solid infill settings for the low half vs upper half of the print - the button needs ironing enabled for the solid infills (don't use for bottom/top layers).  The strap layers don't need ironing

Wiring next... likely won't have time till right after Christmas.  I'm expecting it'll be a minor pain in the butt to get the orientation right so that the cable can exit out the rear.  

I'll need to go back into the CAD and see if I can make it look a bit less like genitals.

![image](https://user-images.githubusercontent.com/24726844/147301555-264f8796-a1d2-44fb-bc6b-b85564584131.png)

**Reply by doudar** [reply]
I wonder if you could square off the middle section in just the lower half (cosmetics) but keep the tops rounded in the upper half (functionality)?

**Comment by eMadman** [comment]
Mod is now up on my fork.  I'll create a PR and merge once I've had a chance to solder the switches and test on my bike.

**Reply by doudar** [reply]
I really like the documentation. Well done. 

And the newest design on the shifter looks very slick as well! 
![image](https://raw.githubusercontent.com/doudar/SmartSpin2k/develop/Hardware/Shifters/screenshot.png)

Merged #287
